catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -19,9 +21,29 @@ import useMemoEq from 'utils/useMemoEq'
 import * as Types from 'utils/types'
 
 import EditFileMeta from './EditFileMeta'
+import EditFileName from './EditFileName'
 import * as PD from './PackageDialog'
 import * as S3FilePicker from './S3FilePicker'
 
+function resolveNameConflictRudely(name: string, attempt: number) {
+  const ext = extname(name)
+  const base = name.split(ext)[0]
+  return `${base} (conflict ${attempt})${ext}`
+}
+
+function resolveName(
+  name: string,
+  obj: Record<string, any>,
+  attempt: number = 1,
+): string {
+  if (!obj[name]) return name
+
+  const newName = resolveNameConflictRudely(name, attempt)
+  if (!obj[newName]) return newName
+
+  return resolveName(name, obj, attempt + 1)
```

Comment:
i doubt that would be an issue in practice, but in theory this can lead to error by exceeding the recursion depth (since js doesn't have a tail-call optimization, unfortunately), so if we are being pedantic, this function should be rewritten using a loop instead of recursion

Reply:
Do you mean you don't want to use recursion in JS in general, or in this particular case?

Reply:
i don't "don't want" to, it's just not safe (may lead to an exception) if the depth is unbounded, and here it well may be, since we don't control what files (and how many) a user adds.
idk, maybe it's kinda ok in practice in this particular case ([source](https://javascript.info/recursion)):

> The maximal recursion depth is limited by JavaScript engine. We can rely on it being 10000, some engines allow more, but 100000 is probably out of limit for the majority of them. There are automatic optimizations that help alleviate this (“tail calls optimizations”), but they are not yet supported everywhere and work only in simple cases.

but in general, in js recursion shouldn't be used for possibly unbounded iterative processes (tho it's ok to use it to traverse recursive data structures, ofc)

Reply:
Ok, but IMO it's an overkill to sacrifice elegance of recursion to cover edge case with 10000 duplicate names. Also, we already have a limit on number of files: 10000